server/bin/pbench-reindex.py

```diff
@@ -0,0 +1,261 @@
+                        continue
+                    # Finally, we have a tar ball that is in the date range
+                    yield c_entry.name, entry.name
+
+
+def main(options):
+    if not options.cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable",
+            file=sys.stderr,
+        )
+        return 1
+
+    try:
+        config = PbenchConfig(options.cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 2
+
+    try:
+        archive_p = Path(config.ARCHIVE).resolve(strict=True)
+    except FileNotFoundError:
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}, does not exist",
+            file=sys.stderr,
+        )
+        return 3
+
+    if not archive_p.is_dir():
+        print(
+            f"The configured ARCHIVE directory, {config.ARCHIVE}, is not a valid directory",
+            file=sys.stderr,
+        )
+        return 4
+
+    try:
+        incoming_p = Path(config.INCOMING).resolve(strict=True)
+    except FileNotFoundError:
+        print(
+            f"The configured INCOMING directory, {config.INCOMING}, does not exist",
+            file=sys.stderr,
+        )
+        return 5
+
+    if not incoming_p.is_dir():
+        print(
+            f"The configured INCOMING directory, {config.INCOMING}, is not a valid directory",
+            file=sys.stderr,
+        )
+        return 6
+
+    _fmt = "%Y-%m-%d"
+    try:
+        oldest_dt = datetime.strptime(options.oldest, _fmt)
+        newest_dt = datetime.strptime(options.newest, _fmt)
+    except Exception as exc:
+        print(
+            f"Invalid time range, {options.oldest} to {options.newest}, "
+            f"'{exc}', expected time range values in the form YYYY-MM-DD",
+            file=sys.stderr,
+        )
+        return 7
+    else:
+        if newest_dt < oldest_dt:
+            # For convenience, swap oldest and newest dates that are reversed.
+            oldest_dt, newest_dt = newest_dt, oldest_dt
+
+    print(f"Re-indexing tar balls in the range {oldest_dt} to {newest_dt}")
+
+    actions = []
+    start = pbench._time()
+    for _val in gen_reindex_list(archive_p, oldest_dt, newest_dt):
+        controller_name, tb_name = _val
+        act_set = reindex(
+            controller_name, tb_name, archive_p, incoming_p, options.dry_run
+        )
+        actions.append(act_set)
+    end = pbench._time()
+
+    for act_set in sorted(actions):
+        print(f"{act_set!r}")
+
+    print(f"Run-time: {start} {end} {end - start}")
+    return 0
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(f"Usage: {_NAME_} [--config <path-to-config-file>]")
+    parser.add_argument("-C", "--config", dest="cfg_name", help="Specify config file")
+    parser.set_defaults(cfg_name=os.environ.get("_PBENCH_SERVER_CONFIG"))
```

Comment:
Incorporate the default setting into the `.add_argument()` call?

Reply:
Fixed.